onnxruntime/core/providers/shared/node_unit/node_unit.h

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <string>
+#include <optional>
+
+#include "core/graph/basic_types.h"
+
+namespace onnxruntime {
+
+class Graph;
+class GraphViewer;
+class Node;
+class NodeArg;
+class Path;
+
+namespace QDQ {
+struct NodeGroup;
+}
+
+class NodeUnit {
+ public:
+  enum class Type : uint8_t {
```

Comment:
documentation for enum members?
or more informative names, like "SingleNode" and "QDQGroup"?